streams/src/test/java/org/apache/kafka/streams/kstream/internals/KStreamSlidingWindowAggregateTest.java

```diff
@@ -328,6 +328,68 @@ public void testAggregateLargeInput() {
         );
     }
 
+    @Test
+    public void testEarlyRecords() {
```

Comment:
Can we add maybe one or two more tests? I think at the least we should have one test that processes _only_ early records, and one test that covers input(s) with the same timestamp.